src/views/Search.vue

```diff
@@ -102,15 +118,18 @@ export default defineComponent({
         event.target.complete();
       })
     },
-    async getProducts(vSize: any, vIndex: any) {
+    async getProducts(vSize: any, vIndex: any, queryString?: string) {
       const viewSize = vSize ? vSize : process.env.VUE_APP_VIEW_SIZE;
       const viewIndex = vIndex ? vIndex : 0;
       const payload = {
         viewSize,
         viewIndex,
         queryString: '*' + this.queryString + '*'
       }
-      if (this.queryString) {
+      if(queryString){
```

Comment:
I think we can simplify it more by applying a ternary operator and defining the value of query String in payload, the same as we have done for viewSize and viewIndex.